go-controller/pkg/util/net_linux.go

```diff
@@ -249,6 +249,67 @@ func LinkAddrFlush(link netlink.Link) error {
 	return nil
 }
 
+// SyncAddresses ensures the link has the provided addresses only
+// Ignores IPv6 LLA
+// addresses should all be of the same family
+func SyncAddresses(link netlink.Link, addresses []*net.IPNet) error {
+	if len(addresses) == 0 {
+		return nil
+	}
+	firstFamily := getFamily(addresses[0].IP)
+	for _, addr := range addresses[1:] {
+		if getFamily(addr.IP) != firstFamily {
+			return fmt.Errorf("all addresses are not the same family: %#v", addresses)
+		}
+	}
+
+	addrs, err := netLinkOps.AddrList(link, firstFamily)
+	if err != nil {
+		return fmt.Errorf("failed to list addresses for the link %s: %v",
+			link.Attrs().Name, err)
+	}
+
+	// desired addresses - true if already exist
+	matched := map[*net.IPNet]bool{}
+	for _, desiredAddr := range addresses {
+		matched[desiredAddr] = false
+	}
```

Comment:
nits: 

you can build this map in the iteration you are already doing above.

but actually it might just be cleaner to build a `map[string]*util.IPNet` of desired addresses. And then handle the difference between desired and actual comparing two extra `k8s.io/apimachinery/pkg/util/sets`.

Reply:
i dont think its worth my time. this code only executes once on start up